### PR TITLE
Daniel Phillips 1705BEE Scrabble Exercise 

### PR DESCRIPTION
**Pull request by dphilla**
PR tested and wrote functioning word scoring class, (i.e. GameReader)

### CODE REVIEW

File: lib/game_reader.rb [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+    "M"=>3, "N"=>1, "O"=>1, "P"=>3,
+    "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+    "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+    "Y"=>4, "Z"=>10
+              }
+  end
+
+   def score(letter)
+    letter = letter.upcase
+    if letter.length > 1
```
Comment (s-espinosa) [inline lib/game_reader.rb]: Interesting! I would flip this on its head and expect `word` instead of `letter` as input. We can still call `#split`, and can do away with the else condition.

File: lib/game_reader.rb [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+      letter = letter.split(//)
+      word_score = letter.map { |character| @score_dict[character]}
+      @score = word_score.reduce(:+)
```
Comment (s-espinosa) [inline lib/game_reader.rb]: Like how clear the thinking is in these three lines. It's all very compact, so I think it's probably fine to keep in a single method, but one thing you might want to consider is creating methods that support this `#score` method that will then allow you to unit test each step along the way.

I could see a version of this with `#split_words`, `#score_letters`, and `#sum_letter_scores` as instance methods that you could call from this `#score` method, and the thinking would continue to be *extremely* clear.

File: lib/game_reader.rb [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+    "E"=>1, "F"=>4, "G"=>2, "H"=>4,
+    "I"=>1, "J"=>8, "K"=>5, "L"=>1,
+    "M"=>3, "N"=>1, "O"=>1, "P"=>3,
+    "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+    "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+    "Y"=>4, "Z"=>10
+              }
+  end
+
+   def score(letter)
```
Comment (s-espinosa) [inline lib/game_reader.rb]: Doesn't look like this handles `nil` or `""`, which we needed per the spec.

File: test/game_reader_test.rb [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+    result = new_game.score("A")
+    assert_equal expected, result
+  end
+
+  def test_can_check_words
+    new_game = GameReader.new
+    expected = 10
+    result = new_game.score("heyo")
+    assert_equal expected, result
+  end
```
Comment (s-espinosa) [inline test/game_reader_test.rb]: Having the support methods mentioned above would allow you to include some unit tests here.

Would also expect tests for `nil` and `""`.
